hip/base/hipsparse_bindings.hip.hpp

```diff
@@ -303,6 +303,128 @@ GKO_BIND_HIPSPARSE_CONJ_TRANSPOSE64(ValueType, detail::not_implemented);
 #undef GKO_BIND_HIPSPARSE_CONJ_TRANSPOSE
 
 
+#define GKO_BIND_HIPSPARSE32_CSRSV2_BUFFERSIZE(ValueType, HipsparseName)     \
+    inline void csrsv2_buffer_size(                                          \
+        hipsparseHandle_t handle, hipsparseOperation_t trans,                \
+        const size_type m, size_type nnz, const hipsparseMatDescr_t descr,   \
+        const ValueType *csrVal, const int32 *csrRowPtr,                     \
+        const int32 *csrColInd, csrsv2Info_t factor_info,                    \
+        int *factor_work_size)                                               \
+    {                                                                        \
+        GKO_ASSERT_NO_HIPSPARSE_ERRORS(HipsparseName(                        \
+            handle, trans, m, nnz, descr,                                    \
+            as_hiplibs_type(const_cast<ValueType *>(csrVal)), csrRowPtr,     \
```

Comment:
Do we know why cuSPARSE/hipSPARSE want non-const access to the values? Why only in the `buffer_size` call?

Reply:
I do not know why they use non-const value, but it should not change the value.